File: week 6/readability.py
a = 1
l = 0  # assigning variables
s = 0
# getting user input
text = (input("text : "))

# designating all the components of readability
for char in text:
    if (char == ' '):
        a += 1

    elif (char == '.' or char == '?' or char == '!'):
        s += 1

    elif (char.isalpha()):
        l += 1

# formula for finding grade
index = round(0.0588 * l/a * 100 - 0.296 * s/a * 100 - 15.8)

# getting output
if (index < 1):
    print("Before Grade 1")

elif (index >= 16):
    print("Grade 16+")

else:
    print("Grade", index)

File: week 2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // inputs
    string text = get_string("text:");
    int let = 0;
    int wrd = 1;
    int sent = 0;
    // loops to identify all the words letters and sentence

    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            let++;
        }
        else if (text[i] == ' ')
        {
            wrd++;
        }
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sent++;
        }
    }
    // getting value for L and S for index
    float L = (float) let / (float) wrd * 100;
    float S = (float) sent / (float) wrd * 100;

    // getting grade from formula
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    // printing grade
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}